registry/storage/signedmanifeststore.go

```diff
@@ -0,0 +1,220 @@
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/manifest/schema1"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/libtrust"
+)
+
+// manifestStore is a storage driver based store for storing schema1 manifests.
```

Comment:
Going with my comment above, I am not sure how this is going to be different from a schema2 store, other than that there won't be a signature fetch. Could we possibly generalize this with a code approach? I'd hate to fork this much code, as it is rather tricky. Albeit, this does seem like it is fairly schema1 specific.

Might be better to call this `schema1ManifestStore`. I think the only real differences are the body of put and get.